cmd/queue/main.go

```diff
@@ -182,6 +196,25 @@ func main() {
 		}(name, server)
 	}
 
+	// Enable TLS server when activator server certs are mounted.
+	// At this moment activator with TLS does not disable HTTP.
+	// See also https://github.com/knative/serving/issues/12808.
+	if tlsEnabled {
+		mainTLSServer, drain := buildServer(ctx, env, probe, stats, logger, concurrencyendpoint, true /* enable TLS */)
+		tlsServers := map[string]*http.Server{
+			"tlsMain":  mainTLSServer,
+			"tlsAdmin": buildAdminServer(logger, drain),
```

Comment:
It feels like moving the admin server to TLS is spread all over this function. What about moving this before the loop on 190 and explicitly moving the server from one map to the other. (It would add another `tlsEnabled` condition to the function, but that feels simpler than the other logic.)

i.e. build a `tlsServer` and `httpServer` map, and then iterate through one or both after building the maps?